### ISSUE TITLE
v1.1.10 stops retrieving data

### ISSUE THREAD

**Issue by PKort** [iprak/yahoofinance#85]
Hi,
I am on version 1.1.9, integration is working correctly. I have netities defined as sensor.yahoofinance_symbol and they are correctly retrieveing values. After upgrade to v1.1.10 integration stops retrieving data anymore. Tried couple of times, the same result.
Kind regards,
PKort

**Comment by iprak** [comment]
Do you see anything related to YahooFinance in the logs? Can you share your setup?

**Comment by pioneershahid** [comment]
same issue here. i see the following in the logs

`Log Details (ERROR)
This error originated from a custom integration.
Logger: custom_components.yahoofinance.coordinator
Source: custom_components/yahoofinance/coordinator.py:197
Integration: Yahoo Finance ([documentation](https://github.com/iprak/yahoofinance), [issues](https://github.com/iprak/yahoofinance/issues))
First occurred: 9:40:34 PM (131 occurrences)
Last logged: 10:56:08 PM

Timed out getting data from https://query1.finance.yahoo.com/v7/finance/quote?symbols=BSX`

 and another error as follows,
`Log Details (WARNING)
This error originated from a custom integration.
Logger: custom_components.yahoofinance.coordinator
Source: custom_components/yahoofinance/coordinator.py:129
Integration: Yahoo Finance ([documentation](https://github.com/iprak/yahoofinance), [issues](https://github.com/iprak/yahoofinance/issues))
First occurred: 9:40:42 PM (131 occurrences)
Last logged: 10:56:08 PM

Error obtaining data, retrying in 20 seconds.`

**Comment by iprak** [comment]
Yes that url looks ok. Can you try some other symbol?

**Comment by pioneershahid** [comment]
Thanks for the reply. I changed the symbol as follows,
`yahoofinance:
   show_trending_icon: true
   #symbols:
   #    - BSX
   symbols:
    - ISTNX
   `
I am getting same error. when i enter `https://query1.finance.yahoo.com/v7/finance/quote?symbols=ISTNX` in the webrowser, it works fine. HA is showing unavailable with following error in the logs.

![Capture](https://user-images.githubusercontent.com/56826511/200573721-2a8ad2a5-7cbc-4def-abe9-3fec8b885712.PNG)

**Comment by iprak** [comment]
That seems bizarre.

You can try manually downgrading by downloading the previous release from https://github.com/iprak/yahoofinance/releases/tag/v1.1.9 and then copying the contents of the zipped folder content `yahoofinance-1.1.9.zip\yahoofinance-1.1.9\custom_components\yahoofinance` into `YourHassio\config\custom_components\yahoofinance`.

**Comment by PKort** [comment]
Yes, downgrade solves the issue. I uninstalled the extension from HACS, and then re-installed it again manually choosing v1.1.9. It is working again. But v1.1.10 is not working for me.

**Comment by pioneershahid** [comment]
> That seems bizarre.
> 
> You can try manually downgrading by downloading the previous release from https://github.com/iprak/yahoofinance/releases/tag/v1.1.9 and then copying the contents of the zipped folder content `yahoofinance-1.1.9.zip\yahoofinance-1.1.9\custom_components\yahoofinance` into `YourHassio\config\custom_components\yahoofinance`.

in my case, it is not working. I am getting same error again and again. i manually download previous versions with same results. not sure what else i need to do to make it work :( happy to create new ticket or issue if you like. thanks

**Comment by iprak** [comment]
I have not been able to reproduce this issue. 1.1.10 did not modify the data load, the request seems to be failing after waiting for 20 seconds which is a mystery. Where is your HomeAssistant instance installed?

**Comment by pioneershahid** [comment]
Hiya, I have HA OS installed on lenovo ThinkCentre mini-PC. I have the system running from last 1 year. not sure if I mentioned previously, this integration was running fine up until Aug/Sep. from last 2-3 months, I started seeing sensor unavailable.

**Comment by iprak** [comment]
Do you see the error all the time? Can you access `https://query1.finance.yahoo.com/v7/finance/quote?symbols=ISTNX` from that machine successfully? I have reached out to Dev community for guidance.

**Comment by pioneershahid** [comment]
I just tried ping the address in the Terminal & SSH addons. i am getting the following error. 

![Capture](https://user-images.githubusercontent.com/56826511/201475406-f6689911-e554-48e7-99ca-676bc9dd477e.PNG)

**Comment by iprak** [comment]
wget indicates that data was correctly downloaded. Right now I am out of ideas right now why the request times out but I will keep poking. The request has 20 second timeout which is quite long.

Can you try one more thing? Navigate to `config\custom_components\yahoofinance` folder on your hass installation. Edit const.py and change query1 to query2 here. And then restart Hass.

`BASE: Final = "https://query1.finance.yahoo.com/v7/finance/quote?symbols="`

**Comment by pioneershahid** [comment]
@iprak unfortunately I am getting same error message as before. 
`
Logger: custom_components.yahoofinance.coordinator
Source: custom_components/yahoofinance/coordinator.py:197
Integration: Yahoo Finance ([documentation](https://github.com/iprak/yahoofinance), [issues](https://github.com/iprak/yahoofinance/issues))
First occurred: 3:01:36 PM (770 occurrences)
Last logged: 10:29:54 PM

Timed out getting data from https://query2.finance.yahoo.com/v7/finance/quote?symbols=ISTNX
`

and 

`
Logger: custom_components.yahoofinance.coordinator
Source: custom_components/yahoofinance/coordinator.py:129
Integration: Yahoo Finance ([documentation](https://github.com/iprak/yahoofinance), [issues](https://github.com/iprak/yahoofinance/issues))
First occurred: 3:01:36 PM (770 occurrences)
Last logged: 10:29:54 PM

Error obtaining data, retrying in 20 seconds.
`

**Comment by pioneershahid** [comment]
@iprak fresh install HA as for some reason when i upgraded OS to 9.3 it caused issue and i was unable to access HA from the web interface. However after fresh install i was hoping it will resolve the issue i was facing. however, i was proved to be wrong and sensor still shows unavailable with error as shown above.

**Comment by lluigi20** [comment]
I am also getting the same error:

Logger: custom_components.yahoofinance.coordinator
Source: custom_components/yahoofinance/coordinator.py:197
Integration: Yahoo Finance (documentation, issues)
First occurred: 11:22:31 AM (39 occurrences)
Last logged: 11:44:21 AM

Timed out getting data from https://query1.finance.yahoo.com/v7/finance/quote?symbols=IVV,^GSPC

I can get the information directly with a browser, so this is not a problem with Yahoo.  I have never been able to get any data through the integration.

Is there any work being done to address this issue?

**Comment by iprak** [comment]
Sorry I have not had any luck replicating this issue on a clean instance. What kind of setup do you have?

**Comment by pioneershahid** [comment]
folks, i had similar issue previously.

with HA supervisor update, I was unable to access HA so i end up doing new installation which did not solve my problem. I was going in circles without resoling it. 

after fresh installation, i was setting up duckdns and trying to setup remote access which was giving me some trouble which is a separate issue. when i remove http section from the configuration.yaml and restarted my pc, yahoofinance started working.  i am not saying removing http solve it but it did sort out my issue with this integration. 

maybe worth giving a shot.. remove http and put yahoofinance after default_config line to see if this works?

**Comment by lluigi20** [comment]
Thanks for the information:  I removed the http: section and restarted but same issue.  I can get the information directly with the browser.
-----------------------------------------------------------------------------------
This error originated from a custom integration.

Logger: custom_components.yahoofinance.coordinator
Source: custom_components/yahoofinance/coordinator.py:197
Integration: Yahoo Finance (documentation, issues)
First occurred: 6:00:57 PM (7 occurrences)
Last logged: 6:04:07 PM

Timed out getting data from https://query1.finance.yahoo.com/v7/finance/quote?symbols=IVV,^GSPC

**Comment by oli737** [comment]
Same problem. After every reboot it update once than never again. Using a PI4. Is there a possible fix for that problem? The entitys nevery update only once when started. I will try later today downgrade the version manually.

**Comment by iprak** [comment]
Please enable debug logging and include the log. The original issue was reported on 1.1.9 which only include the change to flag the sensor as statistics. But 1.1.10 did have some additional adjustments.

**Comment by jboesh** [comment]
The same issue and error here, not able to retrieve data. Ive tried all latest versions.

**Comment by PKort** [comment]
Recently upgraded to version 1.2.1 and it is not working. In the logs I can see following error message:
ERROR (MainThread) [custom_components.yahoofinance.coordinator] Received status 401 (error=Unauthorized) for https://query1.finance.yahoo.com/v7/finance/quote?symbols=EURUSD=X
I tried to open address in browser and I get following:
{"finance":{"result":null,"error":{"code":"Unauthorized","description":"Invalid Crumb"}}}
Does it mean Yahoo has closed the service to retrieve the data?

**Comment by iprak** [comment]
@PKort Maybe, but your issue is the same as https://github.com/iprak/yahoofinance/issues/96.

**Comment by carpii** [comment]
Some stockbroker software I use has made changes to their Yahoo API code recently. This is due to Yahoo now presenting a cookie constent dialog (on the Yahoo Finance website). 

I think Yahoo have recently changed their API so you cannot access v7 until you have agreed to the cookie consent, whereas v6 allowed you to do so.

**Comment by acs-lux** [comment]
Im on v1.2.2 and it worked initially but now I get no updates.
The logs say:
`2023-05-16 11:07:50.343 WARNING (SyncWorker_3) [homeassistant.loader] We found a custom integration yahoofinance which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2023-05-16 11:08:15.286 WARNING (MainThread) [homeassistant.setup] Setup of yahoofinance is taking over 10 seconds.
2023-05-16 11:08:16.920 ERROR (MainThread) [custom_components.yahoofinance.coordinator] Received status 401 (Unauthorized Invalid Cookie) for https://query1.finance.yahoo.com/v7/finance/quote?symbols=VELO-USD,PLU-USD,BTC-EUR,ETH-EUR,HBAR-EUR,ADA-EUR,XRP-EUR,XLM-EUR,ALGO-EUR,BNB-EUR,VET-EUR,VTHO-EUR,CHZ-EUR,QNT-EUR,VRA-EUR,CAKE-EUR,EGLD-EUR,AVAX-EUR,DYDX-USD,CSPR-USD,RDNT-USD&crumb=WbxUXr5FzFt
2023-05-16 11:08:16.934 ERROR (MainThread) [custom_components.yahoofinance.coordinator] Error fetching YahooSymbolUpdateCoordinator data: No data received
2023-05-16 11:08:16.952 WARNING (MainThread) [custom_components.yahoofinance.coordinator] Error obtaining data, retrying in 20 seconds.
2023-05-16 11:11:27.502 WARNING (SyncWorker_18) [homeassistant.loader] We found a custom integration yahoofinance which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant`

**Comment by iprak** [comment]
> Im on v1.2.2 and it worked initially but now I get no updates. The logs say: `2023-05-16 11:08:15.286 WARNING (MainThread) [homeassistant.setup] Setup of yahoofinance is taking over 10 seconds. 2023-05-16 11:08:16.920 ERROR (MainThread) [custom_components.yahoofinance.coordinator] Received status 401 (Unauthorized Invalid Cookie) for https://query1.finance.yahoo.com/v7/finance/quote?symbols=BTC-EUR,ETH-EUR&crumb=WbxUXr5FzFt 2023-05-16 11:08:16.934 ERROR (MainThread) [custom_components.yahoofinance.coordinator] Error fetching YahooSymbolUpdateCoordinator data: No data received 2023-05-16 11:08:16.952 WARNING (MainThread) [custom_components.yahoofinance.coordinator] Error obtaining data, retrying in 20 seconds.`
> 
> Restart has no effect.

What happens if you try to navigate to that url in the log in an incognito browser instance?

**Comment by PKort** [comment]
> What happens if you try to navigate to that url in the log in an incognito browser instance?

{
  "finance": {
    "result": null,
    "error": {
      "code": "Unauthorized",
      "description": "Invalid Cookie"
    }
  }
}

**Comment by iprak** [comment]
Please retest this. I hope that the various changes made since June should have resolved this.
